src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
```

Comment:
I think we need to use generics instead

Reply:
Ye, maybe in this case it's better to make it T[] type not to cast all the time to T[]. But the idea was to make this elementData[] Object type like in the real realization of ArrayList.